src/Analyzers/Core/Analyzers/NamingStyle/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -163,6 +163,7 @@ void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             builder[nameof(NamingStyle)] = applicableRule.NamingStyle.CreateXElement().ToString();
             builder["OptionName"] = nameof(NamingStyleOptions.NamingPreferences);
             builder["OptionLanguage"] = compilation.Language;
+            builder["SymbolSpecificationID"] = applicableRule.SymbolSpecification.ID.ToString();
```

Comment:
I see from [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/NamingStyleDiagnosticAnalyzerBase.cs,139) that for a given symbol, we only report diagnostic for the first rule contains this symbol group.
(i.e. If there are two naming rules using the same symbol group, only the first one would be reported by the analyzer)
So I assume the Guid of the SymbolSpecification should be enough to identify all the symbols for a given rule.